tutor-contrib-multi-plugin/tutor_multi_k8s_plugin/plugin.py

```diff
@@ -6,27 +6,45 @@
 
 from .__about__ import __version__
 
-hooks.Filters.CONFIG_DEFAULTS.add_item(("MULTI_K8S_VERSION", __version__))
-# This plugin assumes you are using a central Traefik load balancer on your
-# cluster but if you want, you could use something else like
-# nginx + cert_manager, though that takes more effort to set up. In any case,
-# this plugin sets up standard k8s Ingress objects on the k8s cluster, so you
-# just need to have some working Ingress controller that can detect them and
-# route traffic to them with automatic HTTPS cert provisioning.
-# Use this MULTI_K8S_INGRESS_CLASS_NAME setting to specify the
-# 'ingressClassName' so that your preferred load balancer will see each
-# Open edX instance.
-hooks.Filters.CONFIG_DEFAULTS.add_item(("MULTI_K8S_INGRESS_CLASS_NAME", "tutor-multi-traefik"))
-hooks.Filters.CONFIG_OVERRIDES.add_items([
-    # Don't use Caddy as a per-instance external web proxy, but do still use it
-    # for per-instance load balancing. Eventually it would be nice to get rid of
-    # it completely as it's not needed in this setup, but that would break
-    # compatibility with existing Tutor k8s plugins that configure Caddy via
-    # patches.
-    ("ENABLE_WEB_PROXY", False),
-    # We are using HTTPS
-    ("ENABLE_HTTPS", True),
-])
+config = {
+    "defaults": {
+        "VERSION": __version__,
+
+        # This plugin assumes you are using either ingress-nginx or Traefik as an
+        # ingress controller to provide you with a central load balancer.
+        # The standard Ingress object uses annotations to trigger the generation
+        # of certificates when using ingress-nginx + cert-manager. See: https://cert-manager.io/docs/usage/ingress/#supported-annotations
+        # In this case the ingress class must include the 'nginx' substring.
+        # In the case of Traefik, there is no such constrain.
+        # provisioned by cert-manager.
+        "INGRESS_CLASS_NAME": "tutor-multi-nginx",
+        # Currently there is no easy way to autodiscover a list of public hosts of your
+        # Open edX installation to add to your ingress routes. This especially evident
+        # when installing additional plugins such as tutor-ecommerce or tutor-minio.
+        # The workaround is to manually add a list of hosts to be routed to the caddy
+        # instance.
+        "INGRESS_HOST_LIST": [ ]
+    },
+    "overrides": {
+        # Don't use Caddy as a per-instance external web proxy, but do still use it
+        # for per-instance load balancing. Eventually it would be nice to get rid of
+        # it completely as it's not needed in this setup, but that would break
+        # compatibility with existing Tutor k8s plugins that configure Caddy via
+        # patches.
```

Comment:
nit/optional: we can remove this comment that says "it would be nice to get rid of it completely" - as discussed [by Regis in this thread](https://discuss.openedx.org/t/deploying-open-edx-on-kubernetes-using-helm/8771/7?u=braden) it is actually performing some useful functions beyond just basic routing of HTTP requests, and it's extremely lightweight.

Reply:
I removed the comment, but I want to revisit the Caddy discussion in the forum at a later date because we recently faced a few issues regarding a high rate of 502 errors when running a large a mount of lms pods.